tests/bluetooth/bsim/mesh/src/test_dfu.c

```diff
@@ -49,6 +51,46 @@ static struct bt_mesh_sar_cfg_cli sar_cfg_cli;
 
 static int dfu_targets_cnt;
 static bool dfu_fail_confirm;
+static bool recover;
+
+static uint64_t blob_id;
+
+static enum stop_phase {
+	DFU_SRV_STOP_NONE,
+
+	/* DFU Server successfully processes Update Start message */
+	DFU_SRV_STOP_START,
+	/* DFU Server receives transfer and stops before image is verified */
+	DFU_SRV_STOP_VERIFY,
+	/* DFU Server verifies image as correct */
+	DFU_SRV_STOP_VERIFY_OK,
+	/* DFU Server verifies image as correct */
+	DFU_SRV_STOP_VERIFY_FAIL,
+	/* DFU Server successfully processes Update Apply message */
+	DFU_SRV_STOP_APPLY,
+	/* FW Update procedure is brought to an end */
+	DFU_SRV_STOP_END,
+} expected_stop_phase;
+
+static uint8_t stop_phase2dfu_phase(enum stop_phase phase)
+{
+	switch (phase) {
+	case DFU_SRV_STOP_NONE:
+		return BT_MESH_DFU_PHASE_IDLE;
+	case DFU_SRV_STOP_START:
+		return BT_MESH_DFU_PHASE_TRANSFER_ACTIVE;
+	case DFU_SRV_STOP_VERIFY:
+		return BT_MESH_DFU_PHASE_VERIFY;
+	case DFU_SRV_STOP_VERIFY_OK:
+		return BT_MESH_DFU_PHASE_VERIFY_OK;
+	case DFU_SRV_STOP_VERIFY_FAIL:
+		return BT_MESH_DFU_PHASE_VERIFY_FAIL;
+	case DFU_SRV_STOP_APPLY:
+		return BT_MESH_DFU_PHASE_APPLYING;
+	default:
+		return BT_MESH_DFU_PHASE_IDLE;
```

Comment:
`enum stop_phase` just replicates `enum bt_mesh_dfu_phase` except some states that are unused. Could we just use it instead?